app/backend/controllers/search.js

```diff
@@ -0,0 +1,23 @@
+const {filterData} = require('../utils')
+
+module.exports.search = async (req, res, next) => {
+    const {Event, TradingEquipment, Article, User} = req.models
+    
+    const term = req.query.q
+
+    if(!term) {
+        return res.status(400).json({errormsg: 'Query string should be passed as query in \'q\'.'})
+    }
+
+    const users = filterData(await User.find().lean(), ['name', 'surname',], term)
+    const events = filterData(await Event.find().lean(), ['Country', 'CalendarId', 'Date', 'Catogory',], term)
+    const tradingEq = filterData(await TradingEquipment.find().lean(), ['code', 'name',], term)
+    const articles = filterData(await Article.find().lean(), ['text', 'title',], term)
+
+    return res.json({
```

Comment:
We do not need to return all attributes of tables. It is enough to return only id, name, surname in users for example.

Likewise, we can only return code of trading equipment. 

Same applies for others too.

After search results, user can visit results specific page.

Reply:
But why don't we return all of them? Is it an overload? Maybe, other teams want to use some other attributes to display them in the search results page. So, it would be nice to return all of the attributes.

Reply:
I'm changing it so it returns only the fields where given keywords are searched.